nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -205,120 +212,54 @@ Costmap2DROS::~Costmap2DROS()
+void Costmap2DROS::setPluginParams(rclcpp::Node::SharedPtr node)
 {
-  RCLCPP_INFO(rclcpp::get_logger("nav2_costmap_2d"), "Loading from pre-hydro parameter style");
-  bool flag;
-  std::string s;
-  std::vector<XmlRpc::XmlRpcValue> plugins;
-
-  XmlRpc::XmlRpcValue::ValueStruct map;
-  SuperValue super_map;
-  SuperValue super_array;
-
-  auto parameters_client = std::make_shared<rclcpp::SyncParametersClient>(nh);
-
-  if (parameters_client->has_parameter("static_map")) {
-    flag = parameters_client->get_parameter<bool>("static_map");
-    if (flag) {
-      map["name"] = XmlRpc::XmlRpcValue("static_layer");
-      map["type"] = XmlRpc::XmlRpcValue("nav2_costmap_2d::StaticLayer");
-      super_map.setStruct(&map);
-      plugins.push_back(super_map);
-
-      auto map_layer = rclcpp::Node::make_shared("static_layer", std::string(nh->get_name()));
-
-      move_parameter(nh, map_layer, "map_topic");
-      move_parameter(nh, map_layer, "unknown_cost_value");
-      move_parameter(nh, map_layer, "lethal_cost_threshold");
-      move_parameter(nh, map_layer, "track_unknown_space", false);
-    }
-  }
-  auto obstacles = rclcpp::Node::make_shared("obstacle_layer", std::string(nh->get_name()));
-  //ros::NodeHandle obstacles(nh, "obstacle_layer");
-
-  if (parameters_client->has_parameter("map_type")) {
-    s = parameters_client->get_parameter<std::string>("map_type");
-    if (s == "voxel") {
-      map["name"] = XmlRpc::XmlRpcValue("obstacle_layer");
-      map["type"] = XmlRpc::XmlRpcValue("nav2_costmap_2d::VoxelLayer");
-      super_map.setStruct(&map);
-      plugins.push_back(super_map);
-
-      move_parameter(nh, obstacles, "origin_z");
-      move_parameter(nh, obstacles, "z_resolution");
-      move_parameter(nh, obstacles, "z_voxels");
-      move_parameter(nh, obstacles, "mark_threshold");
-      move_parameter(nh, obstacles, "unknown_threshold");
-      move_parameter(nh, obstacles, "publish_voxel_map");
-    }
-  } else {
-    map["name"] = XmlRpc::XmlRpcValue("obstacle_layer");
-    map["type"] = XmlRpc::XmlRpcValue("nav2_costmap_2d::ObstacleLayer");
-    super_map.setStruct(&map);
-    plugins.push_back(super_map);
-  }
-
-  move_parameter(nh, obstacles, "max_obstacle_height");
-  move_parameter(nh, obstacles, "raytrace_range");
-  move_parameter(nh, obstacles, "obstacle_range");
-  move_parameter(nh, obstacles, "track_unknown_space", true);
-
-  s = parameters_client->get_parameter<std::string>("observation_sources", std::string(""));
-  std::stringstream ss(s);
-  std::string source;
-  while (ss >> source) {
-    move_parameter(nh, obstacles, source);
-  }
-  move_parameter(nh, obstacles, "observation_sources");
+  std::vector<rclcpp::Parameter> param;
 
-  auto inflation = rclcpp::Node::make_shared("obstacle_layer", std::string(nh->get_name()));
-  //ros::NodeHandle inflation(nh, "inflation_layer");
+  std::vector<std::string> plugin_names = {"static_layer","inflation_layer"};
```

Comment:
I'd add a comment here with obstacle_layer in it as a reminder to add in once this is merged.